cpp/include/Ice/OutgoingResponse.h

```diff
@@ -45,39 +45,37 @@ namespace Ice
          * @param replyStatus The status of the response.
          * @param exceptionId The ID of the exception, when the response carries an exception.
          * @param exceptionMessage The exception message, when the response carries an exception.
-         * @param outputStream The output stream that holds the response. Its underlying buffer is adopted by the new
-         * response.
+         * @param outputStream The output stream that holds the response.
          * @param current A reference to the current object of the request.
          */
         OutgoingResponse(
             ReplyStatus replyStatus,
             std::string exceptionId,
             std::string exceptionMessage,
-            OutputStream& outputStream,
+            OutputStream outputStream,
```

Comment:
We're now moving the output stream into the response (keep it mind you can't copy an output stream).

Reply:
Should we use `&&` to make this clear?

Reply:
I researched this question earlier this week: when to pass by value vs pass by rvalue-reference?

The main use-case for rvalue-reference in signatures is for overloads with const& (copy) and && (move), such as:
https://en.cppreference.com/w/cpp/container/vector/push_back

Without overloading, I would stick with "pass by value".  For a move-only object like OutputStream and OutgoingResponse, it doesn't make a difference (ignoring the cost of move-constructing) since you can only move your object.